src/AppInstallerRepositoryCore/Microsoft/PreIndexedPackageSourceFactory.cpp

```diff
@@ -90,21 +90,35 @@ namespace AppInstaller::Repository::Microsoft
 
                 auto lock = Synchronization::CrossProcessReaderWriteLock::LockForWrite(CreateNameForCPRWL(details));
 
-                UpdateInternal(packageLocation, details, progress);
+                UpdateInternal(packageLocation, packageInfo, details, progress);
             }
 
             void Update(const SourceDetails& details, IProgressCallback& progress) override final
             {
                 THROW_HR_IF(E_INVALIDARG, details.Type != PreIndexedPackageSourceFactory::Type());
 
                 std::string packageLocation = GetPackageLocation(details);
+                Msix::MsixInfo packageInfo(packageLocation);
+
+                // The package should not be a bundle
+                THROW_HR_IF(APPINSTALLER_CLI_ERROR_PACKAGE_IS_BUNDLE, packageInfo.GetIsBundle());
+
+                // Ensure that family name has not changed
+                THROW_HR_IF(APPINSTALLER_CLI_ERROR_SOURCE_DATA_INTEGRITY_FAILURE,
+                    GetPackageFamilyNameFromDetails(details) != Msix::GetPackageFamilyNameFromFullName(packageInfo.GetPackageFullName()));
+
+                if (progress.IsCancelled())
+                {
+                    AICLI_LOG(Repo, Info, << "Cancelling update upon request");
+                    return;
+                }
 
                 auto lock = Synchronization::CrossProcessReaderWriteLock::LockForWrite(CreateNameForCPRWL(details));
 
-                UpdateInternal(packageLocation, details, progress);
+                UpdateInternal(packageLocation, packageInfo, details, progress);
             }
 
-            virtual void UpdateInternal(std::string packageLocation, const SourceDetails& details, IProgressCallback& progress) = 0;
+            virtual void UpdateInternal(const std::string& packageLocation, Msix::MsixInfo& packageInfo, const SourceDetails& details, IProgressCallback& progress) = 0;

```

Comment:
>Msix::MsixInfo& [](http://example.com/codeflow?start=76&length=15)

const?

Reply:
Not without changing all of the methods to also be const correct.  And as this is internal to it I'm not concerned about it not being const.